generators/python-v2/codegen/src/ast/Reference.ts

```diff
@@ -37,6 +37,7 @@ export class Reference extends AstNode {
         this.genericTypes = genericTypes ?? [];
         this.alias = alias;
         this.attribute = attribute ?? [];
+        this.references.push(this);
```

Comment:
This allows a `Reference` object to track itself as a reference. This simplifies lots of other logic such that we can safely look at `AstNode.references` to understand all of a nodes references, rather than add special conditionals in many places to check both `AstNode.references` and check to see whether the node is of type `Reference` and should explicitly be added.